### ISSUE TITLE
On project build, it's adding extra (possibly uncecessary) package references

### ISSUE THREAD

**Issue by define-private-public** [mono/monodevelop#5404]
I have `Newtonsoft.Json` and `NLog` as some NuGet dependencies, so up in my .csproj file, this is here:

```
    <Reference Include="Newtonsoft.Json">
      <HintPath>..\packages\Newtonsoft.Json.11.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
    </Reference>
    <Reference Include="NLog">
      <HintPath>..\packages\NLog.4.5.6\lib\net45\NLog.dll</HintPath>
    </Reference>
```

But sometimes when I build the project, this starts being appended to the References section (multiple times sometimes):
```
   <Reference Include="Newtonsoft.Json">
     <HintPath>bin\Debug\Newtonsoft.Json.dll</HintPath>
   </Reference>
   <Reference Include="NLog">
     <HintPath>bin\Debug\NLog.dll</HintPath>
   </Reference>
```

I have other NuGet packages references, but this doesn't seem to be happening.  Is there any insight into why this happens?

**Comment by mrward** [comment]
Which version of the IDE are you using? (About dialog - Show details) This looks like a problem that used to exist, and was fixed, where if you ran a build you would get extra assembly references added to the project.

**Comment by define-private-public** [comment]
MonoDevelop 7.5 (Build 1255)

**Comment by mrward** [comment]
That git commit hash 07aa1d7917fd5c7b26c3f5f390a9444b7edd887a  does not seem to exist so I cannot check whether that version has the bug. Looking at the 7.5.0.1254 tag, which is close to the version you are using, that seems to have the [unfixed code](https://github.com/mono/monodevelop/blob/monodevelop-7.5.0.1254/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MetadataReferenceCache.cs). This problem is fixed in 7.6.0.947 and later.
